### PR TITLE
Sif packer

### PR DESCRIPTION
**Pull request by ikaneshiro**
Packer that uses the system partition of a SIF file as the rootfs for the bundle and subsequent build.

Attn: @singularityware-admin

### CODE REVIEW

File: src/pkg/build/packer_sif.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+	if err := loopdev.Attach(src, os.O_RDONLY, &number); err != nil {
+		return err
+	}
+
+	if err := loopdev.SetStatus(info); err != nil {
+		return err
+	}
+
+	tmpmnt, err := ioutil.TempDir("/tmp", "tmpmnt-")
```
Comment (tri-adam) [inline src/pkg/build/packer_sif.go]: Is there a reason to explicitly use `/tmp`? I think we should probably leave that blank, that way users can override via `$TMPDIR` if necessary.

File: src/pkg/build/packer_sif.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	//copy filesystem into dest
+	sylog.Debugf("Copying filesystem from %s to %s\n", tmpmnt, dest)
+	cmd := exec.Command("cp", "-r", tmpmnt+`/.`, dest)
+	err = cmd.Run()
+	if err != nil {
+		sylog.Errorf("cp Failed", err.Error())
+		return err
+	}
+
+	return err
```
Comment (tri-adam) [inline src/pkg/build/packer_sif.go]: Nitpick: could be `return nil` here, just makes it a bit clearer that you're exiting without an error at this point.

**Review by tri-adam — CHANGES_REQUESTED** [review]
Two small suggestions. Tested against the new tests in #1774 and I'm not seeing any SIF-related failures there.

**Review by tri-adam — APPROVED** [review]

File: src/pkg/build/packer_sif.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+	return b, nil
+}
+
+// First pass just assumes a single system partition, later passes will handle more complex sif files
+// unpackSIF parses throught the sif file and places each component in the sandbox
+func (p *SIFPacker) unpackSIF(b *Bundle, rootfs string) (err error) {
+
+	// load the container
+	fimg, err := sif.LoadContainer(rootfs, false)
```
Comment (yhcote) [inline src/pkg/build/packer_sif.go:48]: The second parameter to LoadContainer() should be "true", because you do not modify the SIF file.

**Review by yhcote — COMMENTED** [review]
The lookup for the 3 types of FS could be done with GetFromDescr() which would probably shorten the code a bit but I recalled that I did not complete the descriptor specific field lookup. That will come later.

**Pull request merged**